torch/utils/data/datapipes/iter/combining.py

```diff
@@ -60,6 +62,39 @@ def __new__(cls, datapipe, instances):
         return [IterateBuffer(buffer) for i in range(instances)]
 
 
+@functional_datapipe('demux')
+class DemultiplexerIterDataPipe(IterDataPipe):
+
+    def __new__(cls, datapipe, instances, classifier_fn):
+        result = []
+        buffer = list(datapipe)
+
+        def filter_fn(classifier_fn, i, x):
+            return classifier_fn(x) == i
+        return [IterateBuffer(buffer).filter(functools.partial(filter_fn, classifier_fn, i)) for i in range(instances)]
+
+@functional_datapipe('mux')
+class MultiplexerIterDataPipe(IterDataPipe):
```

Comment:
And, how about `roundrobin` or `interleave` as the name of this DataPipe? I think they are more intuitive than `mux`.

Reply:
`mux` and `demux` is kind of coupled names, how about we get more feedback and rename if needed?